Controllers/Ingredient_ComponentsController.cs

```diff
@@ -250,10 +239,9 @@ public ActionResult Delete(int? id)
                 return RedirectToAction("CustomerLogin", "Customers");
             }
 
-            var Ingredients_Components = _recitopiaDbContext.Ingredient_Components.Where(m => m.Id == id).Include(r => r.Ingredients).Where(m => m.Customer_Id == CustomerId).Include(r => r.Components).Where(m => m.Customer_Id == CustomerId);
-
-            // var recipe_Ingredients = recipe_Ingredients_Temp.Where(m => m.Id == id);
+            var Ingredients_Components = await _recitopiaDbContext.Ingredient_Components.Where(m => m.Id == id).Include(r => r.Ingredients).Where(m => m.Customer_Id == CustomerId).Include(r => r.Components).Where(m => m.Customer_Id == CustomerId).ToListAsync();
```

Comment:
You don't need multiple `where` clauses. 

`var Ingredients_Components = await _recitopiaDbContext.Ingredient_Components
     .Include(r => r.Ingredients)
     .Include(r => r.Components).
     .Where(ic => ic.Customer_Id == customerId)
     .ToListAsync();`